packages/promise-kit/index.js

```diff
@@ -3,17 +3,26 @@
 
 /// <reference types="ses"/>
```

Comment:
Now that index.js is growing and has other *.js peers, we need to follow our normal convention of having index.js just be a pass-through, and have all the substantive code in non-index `src/*.js` files.

Reply:
In my opinion, this is still small enough not to need additional ceremony. Both of the modules at this level need entry points in `package.json` because they are exposed API.

Reply:
Which is to say, moving code into `src`  would not eliminate these two files, but hollow both out and leave `export * from './src/x.js` ceremony.

Reply:
I just took a stab at splitting into different files and it's not straightforward, mainly because of the types intermingling and the `BestPipelinablePromise` stuff. Best I could do is move the promise executor stuff into `src`, maybe even the `isPromise` too, but leave the others behind.